sdk/trace/evictedqueue_test.go

```diff
@@ -37,7 +37,7 @@ func TestCopy(t *testing.T) {
 func TestDropCount(t *testing.T) {
 	q := newEvictedQueueEvent(3)
 	var called bool
-	q.logDropped = func() { called = true }
+	q.logDroppedFunc = func() { called = true }
```

Comment:
How about actually testing the logging like in `TestLoggerProviderLogger`?
Then we could get rid of `logDroppedFunc` field.
Would it also not be better performance-wise?

Reply:
@pellared I tried that before I can the unit tests by just having a `logDroppedMsg` but there was no noticeable difference in the benchmark.

Would you like me to change this to `logDroppedMsg` and change the tests to check the logger?

Reply:
It would be nice to have if this is not a problem 😉

Reply:
Hey @pellared 
Please checkout https://github.com/open-telemetry/opentelemetry-go/pull/5858/commits/baad07e574017ebcc38254b0228b2470f3ef1dc2 and let me know what you think :smile: